client/layout/controller.js

```diff
@@ -53,11 +54,30 @@ class Controller extends Component {
 		const search = this.props.location.search.substring( 1 );
 		const query = parse( search );
 		const page = find( getPages(), { path } );
+		window.wpNavMenuUrlUpdate( page.wpMenu, query );
 		window.wpNavMenuClassChange( page.wpMenu, this.props.location.pathname );
 		return createElement( page.container, { params, path: url, pathMatch: path, query } );
 	}
 }
 
+// Update links in wp-adming menu to persist time related queries
+window.wpNavMenuUrlUpdate = function( menuClass, { period, compare, after, before } ) {
+	const timeRelatedQuery = omitBy( { period, compare, after, before }, isUndefined );
+	const search = stringifyQuery( timeRelatedQuery );
+
+	Array.from( document.querySelectorAll( `#${ menuClass } a` ) ).forEach( item => {
+		/**
+		 * Example href:
+		 *
+		 * http://example.com/wp-admin/admin.php?page=wc-admin#/analytics/orders?period=today&compare=previous_year
+		 */
+		const hashUrl = last( item.href.split( 'wc-admin#' ) );
+		const base = hashUrl.split( '?' )[ 0 ];
+		const href = '/wp-admin/admin.php?page=wc-admin#' + base + search;
```

Comment:
That fails if WP is installed in a subdirectory (like `http://example.com/example-subdirectory/wp-admin/admin.php?page=wc-admin#...`).

Didn't do a lot of testing, but that seemed to work in my local installation:
```ES6
		const url = item.href.split( 'wc-admin#' );
		const hashUrl = last( url );
		const base = hashUrl.split( '?' )[ 0 ];
		const href = url[ 0 ] + 'wc-admin#' + base + search;
```

Reply:
@Aljullu 's option also worked for me on a local install in a subdirectory.

Reply:
Good catch @Aljullu !